libres/lib/analysis/update.cpp

```diff
@@ -66,7 +67,7 @@ typedef struct {
         *node_info; /* mutable: For the serialization of one node */
 } serialize_info_type;
 
-class update_data_type {
+class update_data_type : public std::enable_shared_from_this<update_data_type> {
```

Comment:
Why do you use `std::shared_ptr<>` here? My experience is that use of smart pointers in the api reduces the generality; maybe what you are after can be achieved with a suitable return  value policy when binding the functions for Python?